subprojects/files/src/main/java/org/gradle/api/internal/file/archive/impl/JdkZipEntry.java

```diff
@@ -14,20 +14,22 @@
  * limitations under the License.
  */
 
-package org.gradle.api.internal.file.archive;
+package org.gradle.api.internal.file.archive.impl;
 
 import com.google.common.io.ByteStreams;
+import org.gradle.api.internal.file.archive.ZipEntry;
+import org.gradle.internal.file.FileException;
 
 import java.io.IOException;
 import java.io.InputStream;
-import java.util.function.Supplier;
+import java.util.concurrent.Callable;
 
 class JdkZipEntry implements ZipEntry {
 
     private final java.util.zip.ZipEntry entry;
-    private final Supplier<InputStream> inputStreamSupplier;
+    private final Callable<InputStream> inputStreamSupplier;
 
-    public JdkZipEntry(java.util.zip.ZipEntry entry, Supplier<InputStream> inputStreamSupplier) {
+    public JdkZipEntry(java.util.zip.ZipEntry entry, Callable<InputStream> inputStreamSupplier) {
```

Comment:
I wouldn't change this. Maybe we should introduce something like `IOSupplier` in `:files` that would throw `IOException`? But I'd just leave this untouched otherwise.

Reply:
Same issue - `Supplier` is only available in JDK8.  I did mention this in the PR description, but maybe you didn't see it.  

I was just looking for a pre-JDK8 equivalent, but I forgot that there was a `Supplier` in guava, I'll just use that instead.

Reply:
👍